components/profile/Profile.tsx

```diff
@@ -0,0 +1,145 @@
+import ProfileBanner from '../../components/profile/ProfileBanner';
+import {
+  HStack,
+  Center,
+  VStack,
+  useColorModeValue,
+  Box,
+  Button,
+  Icon,
+  ScrollView,
+} from 'native-base';
+import { User } from '../../xplat/types/user';
+import { useState } from 'react';
+import { useEffect } from 'react';
+import StatBox from '../../components/profile/StatBox';
+import { Post as PostObj } from '../../xplat/types/post';
+import { Ionicons } from '@expo/vector-icons';
+import Tintable from '../../components/util/Tintable';
+import { Pressable } from 'native-base';
+import Feed from '../media/Feed';
+import { useRecoilValue } from 'recoil';
+import { userAtom } from '../../utils/atoms';
+
+/**
+ * The profile component displays the profile banner, a statbox,
+ * a button to see followers and the post's the user has posted.
+ *
+ * If the prop profileIsMine is true then there will be button
+ * to edit the profile. If it's false then it will be a follow
+ * button.
+ */
+type Props = {
+  profileIsMine: boolean;
+  userOfProfile: User;
+  navigate: Function;
+};
+const Profile = ({ profileIsMine, userOfProfile, navigate }: Props) => {
+  const [boulderGrade, setBoulderGrade] = useState<string>('');
+  const [topRopeGrade, setTopRopeGrade] = useState<string>('');
+  const [numOfSends, setNumOfSends] = useState<string>('');
+  const [posts, setPosts] = useState<PostObj[]>();
```

Comment:
Lets default initialize this to `[]` instead of `undefined`
```suggestion
  const [posts, setPosts] = useState<PostObj[]>([]);
```